### PR TITLE
feat(app-platform): Add all webhook events to request log in dashboard

### PR DESCRIPTION
**Pull request by nolachen**
Extend #15269 by adding all possible webhook request types to the Redis buffer

In this PR I am adding:
- event_alert.triggered
- installation.created
- installation.deleted
- select.requested
- external_issue.created
- external_issue.linked

### CODE REVIEW

**Comment by scefali** [comment]
@nolachen I am not a big fan of the nomenclature for the new issue events. I don't think the `uri` bit is intuitive. And the `request` part is also odd. How about:
`external_issue.created`
`external_issue.linked`

File: src/sentry/mediators/external_requests/util.py [hunk 1]
```diff
@@ -35,3 +40,29 @@ def validate(instance, schema_type):
         return False
 
     return True
+
+
+def send_and_save_request(sentry_app, org_id, event, **kwargs):
```
Comment (scefali) [inline src/sentry/mediators/external_requests/util.py]: @nolachen Why is `url` an optional parameter?

Comment (nolachen) [reply src/sentry/mediators/external_requests/util.py]: @scefali Fixed, it should not be

File: src/sentry/mediators/external_requests/util.py [hunk 2]
```diff
@@ -35,3 +40,26 @@ def validate(instance, schema_type):
         return False
 
     return True
+
+
+def send_and_save_request(sentry_app, org_id, event, url, **kwargs):
```
Comment (scefali) [inline src/sentry/mediators/external_requests/util.py]: @nolachen I think something in the name of this function should reflect that it's only used for sentry apps.

Comment (nolachen) [reply src/sentry/mediators/external_requests/util.py]: @scefali I thought that everything in /mediators is for Sentry Apps? Or is that not true

Comment (scefali) [reply src/sentry/mediators/external_requests/util.py]: @nolachen Nope, we have mediators for plugins and other things as well.

File: src/sentry/tasks/sentry_apps.py [hunk 3]
```diff
@@ -274,29 +272,27 @@ def send_webhooks(installation, event, **kwargs):
...
-            response_code=resp.status_code,
-            org_id=installation.organization_id,
-            event=event,
-            url=servicehook.sentry_app.webhook_url,
+        send_and_save_webhook_request(
+            servicehook.sentry_app.webhook_url, installation.sentry_app, request_data
         )
+
+
+def send_and_save_webhook_request(url, sentry_app, app_platform_event):
```
Comment (scefali) [inline src/sentry/tasks/sentry_apps.py]: @nolachen I'd like to be consistent about the order of arguments for some of these functions. For example, `url` before `sentry_app` here but in `send_and_save_request` it's after.

File: src/sentry/mediators/external_requests/util.py (hunk 2 above)
Comment (nolachen) [reply src/sentry/mediators/external_requests/util.py]: @scefali just to make sure is `select_requester` and `issue_link_requester` just for Sentry Apps though?

Comment (scefali) [reply src/sentry/mediators/external_requests/util.py]: @nolachen That's true. But even if the mediators were only used for SentyrApps, there's nothing in the path that says that so someone who doesn't know that part of the code wouldn't know that.

File: src/sentry/tasks/sentry_apps.py (hunk 3 above)
Comment (nolachen) [reply src/sentry/tasks/sentry_apps.py]: @scefali Yeah I agree. Also if you know a better place (maybe like a utils file?) where this function could belong that might make more sense too

File: src/sentry/mediators/external_requests/select_requester.py [hunk 4]
```diff
@@ -50,7 +50,15 @@ def _build_url(self):
 
     def _make_request(self):
         try:
-            body = safe_urlread(safe_urlopen(url=self._build_url(), headers=self._build_headers()))
+            body = safe_urlread(
+                send_and_save_sentry_app_request(
+                    self._build_url(),
+                    self.sentry_app,
+                    self.install.organization_id,
+                    "select.requested",
```
Comment (scefali) [inline src/sentry/mediators/external_requests/select_requester.py]: @nolachen Nit: What about calling this `select_fields.requested`? I am concerned that `select.requested` might be a bit too ambiguous.

File: src/sentry/tasks/sentry_apps.py (hunk 3 above)
Comment (scefali) [reply src/sentry/tasks/sentry_apps.py]: @nolachen I think the current location is OK.

File: src/sentry/mediators/external_requests/select_requester.py (hunk 4 above)
Comment (nolachen) [reply src/sentry/mediators/external_requests/select_requester.py]: @scefali would it be confusing in that the request is just for 1 select field? Maybe `select_field.requested` ? or like `select_options.requested`?

Comment (scefali) [reply src/sentry/mediators/external_requests/select_requester.py]: @nolachen I like `select_options.requested`

**Review by scefali — APPROVED** [review]

**Pull request merged**